source/parser/util.civet

```diff
@@ -563,18 +563,33 @@ function wrapIIFE(expressions: StatementTuple[], asyncFlag?: boolean): ASTNode
   signature :=
     modifier:
       async: !!async.length
+      generator: !!generator
     returnType: undefined
 
-  fn := makeNode {
-    type: "ArrowFunction"
-    signature
-    parameters
-    returnType: undefined
-    ts: false
-    async
-    block
-    children: [async, parameters, "=>", block]
-  }
+  let fn
+  if generator
+    fn = makeNode {
+      type: "FunctionExpression"
```

Comment:
One thing we'll want to do in the future (doesn't need to hold up this PR) is to bind `this` and shadow arguments if necessary.

`.bind(this)()`

To detect if we should shadow arguments we can look inside the function to see if there is an `arguments` identifier that isn't declared and if so pass it in like:

```
if needsToShadowArguments
  parameters :=
    type: "Parameters"
    children: ["(arguments)"]
    names: ["arguments"]
  
  ...
  
  .bind(this)(arguments)

```

Thanks for the PR! We're always happy to have more contributors. Again, this doesn't need to hold anything up, just wanted to note it while I was thinking about it and if it doesn't make it in this one then one of us can get to it eventually.